components/primitives/Button/Button.tsx

```diff
@@ -0,0 +1,50 @@
+import { cn } from "@/lib/utils/cn";
+import React, { forwardRef } from "react";
+
+interface ButtonProps extends React.ComponentPropsWithRef<"button"> {
+  className: string;
+  disabled?: boolean;
+  fullWidth?: boolean;
+  children: React.ReactNode;
+  size?: "sm" | "base" | "lg";
+  variant?: "primary" | "secondary" | "danger";
+}
+
+const Button = (
+  {
+    className,
+    children,
+    fullWidth,
+    variant,
+    size,
+    disabled,
+    ...props
+  }: ButtonProps,
+  ref: React.ForwardedRef<HTMLButtonElement>,
+) => {
+  const buttonClass = {
+    "bg-abeg-error-20": variant === "danger",
+    "border-abeg-green text-abeg-green border": variant === "secondary",
+    "bg-abeg-neutral-20": variant === "primary",
+    "px-3": size === "sm",
+    "px-10": size === "lg",
+    "w-full": fullWidth,
+    "bg-abeg-neutral-50 text-white": disabled,
+  };
+
+  return (
+    <button
+      className={cn(
+        "px-5 py-2 rounded-lg text-white text-sm font-medium",
+        buttonClass,
```

Comment:
Impressive what I see here 🤓

But, I'm curious to know how the `className` is resolved from the usage of the `buttonClass` object.

If you got a resource you wanna share that explains this approach, do share or If you want to lesson me too, I'm down. 

🤲🏾

Reply:
It uses the varadic object format (which apparently you love so much🌝), which clsx accepts and resolves to the className strings gotten from the object keys and long as the condition placed as the object values🤝

Reply:
> Impressive what I see here 🤓
> 
> But, I'm curious to know how the `className` is resolved from the usage of the `buttonClass` object.
> 
> If you got a resource you wanna share that explains this approach, do share or If you want to lesson me too, I'm down. 
> 
> 🤲🏾

Yeah I just used the clsx docs to understand the package.